### PR TITLE
Deploy resources

### PR DESCRIPTION
**Pull request by NickGraham101**
### Context

We need to be able to deploy resources and code for this application.

### Changes proposed in this pull request

- Add makefile
- Add workflows to build and deploy app
- Add terraform to define infrastructure
- Bump Ruby version in Dockerfile and add some other fixes
- Update database names to reflect new project name
- Enable ssh access to container

### Guidance to review

- Check test run [logs](https://github.com/DFE-Digital/access-your-teaching-qualifications/actions/runs/4292261695) in GitHub Actions
- Check resources in s165d01-aytq-dev-rg are correct

### Link to Trello card

https://trello.com/c/eT6yaaHD

### Checklist

- [ ] Attach to Trello card
- [ ] Rebased main
- [ ] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

**Comment by NickGraham101** [comment]
[Infrastructure and domain naming](https://trello.com/c/eT6yaaHD/694-infrastructure-and-domain-naming)

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+	$(eval ENV_TAG=dev)
+
+.PHONY: review
+review: review-init set-azure-resource-group-tags
+	$(eval DEPLOY_ENV=review)
+	$(eval AZURE_SUBSCRIPTION=s165-teachingqualificationsservice-development)
+	$(eval RESOURCE_NAME_PREFIX=s165d01)
+	$(eval ENV_SHORT=rv)
+	$(eval env=-pr-$(pr_id))
+	$(eval backend_config=-backend-config="key=review/review$(env).tfstate")
```
Comment (smneal) [inline Makefile]: Should we be standardising the case of these to match all the others?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+	bin/fetch_config.rb -s azure-key-vault-secret:${KEY_VAULT_NAME}/${KEY_VAULT_SECRET_NAME} -f yaml
+
+validate-keyvault-secret: read-keyvault-config install-fetch-config set-azure-account
+	bin/fetch_config.rb -s azure-key-vault-secret:${KEY_VAULT_NAME}/${KEY_VAULT_SECRET_NAME} -d quiet \
+		&& echo Data in ${KEY_VAULT_NAME}/${KEY_VAULT_SECRET_NAME} looks valid
+
+terraform-init:
+	$(if $(IMAGE_TAG), , $(eval export IMAGE_TAG=main))
+	[[ "${SP_AUTH}" != "true" ]] && az account set -s $(AZURE_SUBSCRIPTION) || true
+	terraform -chdir=terraform init -backend-config workspace_variables/${DEPLOY_ENV}.backend.tfvars $(backend_config) -upgrade -reconfigure
```
Comment (smneal) [inline Makefile]: Should we standardise variable references to either ${} or $() where appropriate?

File: Makefile (hunk 1 above)
Comment (NickGraham101) [reply Makefile]: Good spot, done

**Review by smneal — APPROVED** [review]
LGTM

File: Makefile (hunk 2 above)
Comment (NickGraham101) [reply Makefile]: replaced $() with ${} (except make commands)

**Pull request merged**